### ISSUE TITLE
Google Photos API 429 Too Many Requests

### ISSUE THREAD

**Issue by BigThinker** [mvysny/photocloud-frame-slideshow#134]
I get a 429 error (Too Many Requests) when using the slideshow from Google Photos. With what account are the photos fetched, because I couldn’t see any related project in my Google Developer Console. I tried to check the Quota tab because apparently there’s a 75000 requests limit per project per day which, as I calculated, with 5 seconds intervals photos slideshow shouldn’t surpass the limit.

**Comment by mvysny** [comment]
Hi, thank you for letting me know! Yeah, unfortunately the quotas are project-specific, rather than user-specific. That means that the following quota is valid for all PhotoCloud users:

```
The quota limit for requests to the Library API is 10,000 requests per project per day.

The quota limit for requests to access media bytes (by loading a photo or video from a base URL) is 75,000 requests per project per day.
```

https://developers.google.com/photos/library/guides/api-limits-quotas

I need to figure out whether it's possible to raise the quota limit...

**Comment by mvysny** [comment]
There is a partner's program which may increase the quota here: https://developers.google.com/photos/partner-program/overview I'll apply, let's see where that will lead to.

**Comment by mvysny** [comment]
I've applied, send the screenshots, waiting for Google to review and reply.

**Comment by mvysny** [comment]
I've hopefully became a Google Partner, which should lift the limits quite a bit. Can you please confirm that you're no longer getting this issue, or are you still getting this bug?

**Comment by naisanzaa** [comment]
> I've hopefully became a Google Partner, which should lift the limits quite a bit. Can you please confirm that you're no longer getting this issue, or are you still getting this bug?

did you get approved for the program?

**Comment by mvysny** [comment]
Google is now reviewing my request for public access, but it will take some time. Quoting google:

> External apps that request sensitive or restricted user data must first be verified by Google. This process may take 4-6 weeks

Let’s hope that the process will take faster than 4-6 weeks.

Once the app is approved, I'll release a new version of PhotoCloud to Google Play.

**Comment by mvysny** [comment]
Fuck. I figured out what the problem is.

In order for PhotoCloud to access Google Drive/Google Photos, you need to create a project in Google Cloud, then head to APIs and create a Oauth 2.0 Client ID, which then links the Google Cloud access with PhotoCloud.

The problem is that I did that with my old e-mail address martin.vysny2@gmail.com. I had to nuke that e-mail since its in-app payments/billing profile was registered to Slovakia, and I moved to Finland but Google is unable to change the billing country and so I had to create a new user (!!!!!).

However, apparently Google didn't delete everything, because now when I try to create the OAuth 2.0 Client ID in my new project owned by martin.vysny3@gmail.com, Google fails with:

```
Ownership verification failed

The request failed because the Android package name and fingerprint are already in use

Request ID: 6499439294717751697
```

When I googled for the error message, people suggested to change the fingerprint, but I don't want to do that since I have a huge existing user base and the app could break for them.

I tried to get help at Google Cloud but unless I pay big buck, I'll hit an unpenetrable wall of chat-bots that just ask their canned questions, which is completely unhelpful.

Then I found https://support.google.com/firebase/answer/6401008 and it told me to reach to Firebase support centre, so I'm trying to do that, but I don't have high hopes on solving this.

The bottom line is that Google fucking sucks balls. I just wasted 3 hours trying to fight the system, and I can't reach any human operator to help me out. Google documentation is utterly useless - it's long and contains zero information, one needs an AI to parse something useful out of that crap.

**Comment by mvysny** [comment]
I personally switched away from Google completely a year ago. I've switched to the Apple ecosystem, threw away my Android and got iPhone; moved my e-mail to Apple iCloud (and it was quite easy to do ). I urge you to do the same thing - the sense of privacy is much better than with Google (remember, it's Google business to sell your data). The Apple Photo app is much better and is very well integrated - it's easy to get a working photo frame.

I'll keep maintaining PhotoCloud but I won't waste my time much - so no new features will ever be incoming to PhotoCloud. If Google throws some big stick under my feet, I'll abandon PhotoCloud completely. Life is too short to waste fighting with fucking Google.

**Comment by mvysny** [comment]
It's possible to upgrade the app signing key on Google Play Console:

```
Upgrade app signing key

This upgrade will apply your new key on Android N (API level 24) and above.
New and existing installs on older Android releases will ignore the new key until they update to Android N or above. [Learn more](https://support.google.com/googleplay/android-developer/answer/9842756?hl=en#upgrade&zippy=request-a-key-upgrade-for-all-installs-on-android-n-api-level-and-above) 
```
This sounds like something that should hopefully preserve all Google services like IAP and stuff.

**Comment by mvysny** [comment]
Okay, I've changed the signing key for the app and created the OAuth Client ID for the app, and the change went through. Now waiting for Google to approve the OAuth consent screen in order to go public.

**Comment by mvysny** [comment]
Still waiting for Google: "Verification Status: Verification in progress"
